.github/workflows/test.yaml

```diff
@@ -73,17 +73,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - name: Set up Python 3.11
-        uses: actions/setup-python@v5
-        with:
-          python-version: "3.11"
-          cache: "pip"
-          cache-dependency-path: "**/pyproject.toml"
-      - name: set python default branch
+      - name: Install Hatch
+        uses: pypa/hatch@install
```

Comment:
https://github.com/pypa/hatch/tree/install
Kinda weird that they don't versionize it properly but instead recommend to pin a commit. It would be better but look a bit ugly. I would hope that such a simple action is stable enough.